src/Microsoft.AspNet.Razor/Editor/SpanEditHandler.cs

```diff
@@ -169,17 +170,16 @@ public override string ToString()
         public override bool Equals(object obj)
         {
             var other = obj as SpanEditHandler;
-            return other != null &&
-                   AcceptedCharacters == other.AcceptedCharacters &&
-                   EditorHints == other.EditorHints;
+            return obj != null &&
```

Comment:
why?  if the `GetType()` comparison succeeds, the conditional cast must also succeed.  this allows the compiler to skip the cast attempt when one of the other comparisons fails.